src/services/courseService.ts

```diff
@@ -0,0 +1,67 @@
+import axiosInstance from './axiosInstance';
+import {isAxiosError} from 'axios';
+
+export type CourseDetailsDTO = {
+    id: string;
+    city: string;
+    compact: boolean;
+    country: string;
+    golfName: string;
+    name: string;
+    numHoles: number;
+    pitchAndPutt: boolean;
+    postalCode: string;
+};
+
+const fetchData = async (url: string, params: object = {}) => {
+    try {
+        const {data} = await axiosInstance.get(url, {params});
+        return data;
+    } catch (error) {
+        if (isAxiosError(error) && error.response?.status === 404) {
+            throw new Error('Resource not found (404)');
+        }
+        throw new Error('Error fetching data');
+    }
+};
+
+export const getCourseDetails = (page: number = 1, pageSize: number = 12) =>
+    fetchData('/v1/courses/details', {page, pageSize});
+
+export const searchCourseDetails = (searchTerm: string, page: number = 1, pageSize: number = 12) =>
+    fetchData('/v1/courses/details/search', {searchTerm, page, pageSize});
+
+export const getCourseDetailsById = (id: string) =>
+    fetchData(`/v1/courses/${id}/details`);
+
+const updateData = async (url: string, payload: object) => {
```

Comment:
If you find yourself having a function with no reference to the current file, you may find it better fits in another file.  This is much better placed in your axiosInstance.ts file :)